### ISSUE TITLE
Create design for My Achievements page

### ISSUE THREAD

**Issue by Abhijay007** [codeforcauseorg/edu-client#292]
## Is your feature request related to a problem? Please describe.

A My achievements page where users can see his/her rankings and achievement level, discussion answered, assignments competed, his/her top scores, badges, etc.

## Describe the solution you'd like

Create a page UI for my achievement section under the profile section.

**Comment by Abhijay007** [comment]
@Abhishek-kumar09 can I work on this?

**Comment by vasudev19126** [comment]
@Abhijay007 Yeah, go ahead.

**Comment by rohank2502** [comment]
Hey Please add space to showcase badges also!

**Comment by Abhijay007** [comment]
> Hey Please add space to showcase badges also!

ok, I will also add that 😊

**Comment by adarsh-technocrat** [comment]
Hey, @Abhijay007 I would suggest making a Figma design first because that will reduce any kind of rework right and get approval from @Abhishek-kumar09, @vasudevsinghal.

and if you need any kind of UI suggestion or help in the design part you can ping me anywhere. 😃

**Comment by Abhijay007** [comment]
> Hey, @Abhijay007 I would suggest making a Figma design first because that will reduce any kind of rework right and get approval from @Abhishek-kumar09, @vasudevsinghal.
> 
> and if you need any kind of UI suggestion or help in the design part you can ping me anywhere. 😃

Ya! sure @adarsh-technocrat you can share the Figma design here I will try my best to implement code accordingly. We can discuss ideas and changes in UI in this thread 😊

**Comment by adarsh-technocrat** [comment]
sure @Abhijay007 I'll let you know!

**Comment by adarsh-technocrat** [comment]
Hey, @Abhijay007  can you please list down the things that we need to put in the My Achievement section! One thing already mentioned is showcase badges.

**Comment by Abhijay007** [comment]
> Hey, @Abhijay007 can you please list down the things that we need to put in the My Achievement section! One thing already mentioned is showcase badges.

ok, @adarsh-technocrat so I think that first on the top we can make a section  where we can show profile pic,followers, and current achievement level, something like this:

![main profiler](https://user-images.githubusercontent.com/64387054/117400135-ab6fa600-af1f-11eb-9e5a-0d0ace763cd8.PNG)

followed by this we can add a div section in which we can show details as courses completed, discussion answered, a=bages earned something like this :

![detail div](https://user-images.githubusercontent.com/64387054/117400207-c510ed80-af1f-11eb-8c10-4bce31611c07.PNG)

and after this we can showcase some rare badges on the a page like this by making a separate section like this: 

![badges](https://user-images.githubusercontent.com/64387054/117400350-06090200-af20-11eb-8eb1-e3dc0671ea6f.PNG)

and by clicking on the badges we will redirect the user to a dedicated badges page where the user can see badges descriptions and details.


PS: ignore the color scheme for now 😄 and @adarsh-technocrat if you have more suggestions you can add them. I am also thinking about other sub-sections.

**Comment by adarsh-technocrat** [comment]
Great! @Abhijay007 sounds good I'll make something according to my imagination including your lovely suggestions and further we have different iterations on it as I complete the UI  :)

let @Abhishek-kumar09  @vasudevsinghal  also put their thoughts on the same!

**Comment by Abhijay007** [comment]
> Great! @Abhijay007 sounds good I'll make something according to my imagination including your lovely suggestions and further we have different iterations on it as I complete the UI :)
> 
> let @Abhishek-kumar09 @vasudevsinghal also put their thoughts on the same!

Ya! sounds good!!

**Comment by adarsh-technocrat** [comment]
Hey, @Abhijay007  actually I'm designing the UI here is the [Figma link](https://www.figma.com/file/pm7EIe6PM9j48fannjv8kr/My-Acheviment?node-id=0%3A1)  where you can see it :)

**Comment by Abhijay007** [comment]
> Hey, @Abhijay007 actually I'm designing the UI here is the [Figma link](https://www.figma.com/file/pm7EIe6PM9j48fannjv8kr/My-Acheviment?node-id=0%3A1) where you can see it :)

ok, @adarsh-technocrat got it !! looks good 👍

**Comment by adarsh-technocrat** [comment]
Hey, @Abhijay007 I think my achievement can be part of the gamification section because the things that we are putting in my achievement section it's a bit similar to those required in gamification section  #166 what your thoughts?

Cc/ @Abhishek-kumar09

**Comment by Abhijay007** [comment]
> Hey, @Abhijay007 I think my achievement can be part of the gamification section because the things that we are putting in my achievement section it's a bit similar to those required in gamification section #166 what your thoughts?
> 
> Cc/ @Abhishek-kumar09

yes, @adarsh-technocrat these are quite similar but I believe that these issues can be break up into different PRs like we can make a separate leaderboard section instead of showing it as a part of the gamification or my achievements page and we can show raking of an individual on achievements page and we may also need to make a separate badge page to showcase an individual's performance. what I want to say that we can break this issue into multiple PRs and starting working on, each one by one for that we can set a **GitHub milestone** and try to discuss the various goals we need to achieve in order to accomplish this task. we can discuss various possibilities and ideas there to make fully functional gamification or my achievements page like we may need a lot of ideas here, we need to make a whole Figma file include UI and need to break down that into smaller issues and PRs just like we did for main/basic design in starting of the project. 

what are your thoughts on this?

**Comment by adarsh-technocrat** [comment]
yaa! @Abhijay007 I think let mentor clear this thing because until and unless we get familiar with the workflow of the application we cant take decisions right! and the UI that I'm making is a bit off track as per the theme of the application I need to change that.
sorry for taking time for this :)

**Comment by Abhijay007** [comment]
> > Hey, @Abhijay007 I think my achievement can be part of the gamification section because the things that we are putting in my achievement section it's a bit similar to those required in gamification section #166 what your thoughts?
> > Cc/ @Abhishek-kumar09
> 
> yes, @adarsh-technocrat these are quite similar but I believe that these issues can be break up into different PRs like we can make a separate leaderboard section instead of showing it as a part of the gamification or my achievements page and we can show raking of an individual on achievements page and we may also need to make a separate badge page to showcase an individual's performance. what I want to say that we can break this issue into multiple PRs and starting working on, each one by one for that we can set a **GitHub milestone** and try to discuss the various goals there about what we need to achieve in order to accomplish this task. we can discuss various possibilities and ideas there to make fully functional gamification or my achievements page like we may need a lot of ideas here, we need to make a whole Figma file include UI and need to break down that into smaller issues and PRs just like we did for main/basic design in starting of the project.
> 
> what are your thoughts on this?

@Abhishek-kumar09 can I implement this?

**Comment by rajatgupta24** [comment]
> > Hey, @Abhijay007 I think my achievement can be part of the gamification section because the things that we are putting in my achievement section it's a bit similar to those required in gamification section #166 what your thoughts?
> > Cc/ @Abhishek-kumar09
> 
> my achievements page and we can show raking of an individual on achievements page and we may also need to make a separate badge page to showcase an individual's performance.

I also agree with @Abhijay007, On the achievements page, we can show the overall ranking of the user & things that can help him in improving it, whereas, on the badges page, we can show small tasks that users completed. I created a UI for badge page #239

**Comment by Abhijay007** [comment]
@rajatgupta24 , @adarsh-technocrat is working on a new UI design we can make PR according to that we are using a different approach this time, we are making a single profile section in which achievements page has a separate scrollable section.

**Comment by Abhijay007** [comment]
> I also agree with @Abhijay007 & @adarsh-technocrat, On the achievements page, we can show the overall ranking of the user & things that can help him in improving it, whereas, on the badges page, we can show small tasks that users completed. I created a UI for badge page #239

@rajatgupta24 , @adarsh-technocrat is working on a new UI design we can make PR according to that we are using a different approach this time, we are making a single profile section in which achievements page has a separate scrollable section.
